testing/src/scenario/runtime.py

```diff
@@ -254,7 +253,7 @@ def _get_event_env(self, state: "State", event: "_Event", charm_root: Path):
                 else:
                     logger.warning(
                         "remote unit ID unset; no remote unit data present. "
-                        "Is this a realistic scenario?",  # TODO: is it?
+                        "Is this a realistic scenario?",
```

Comment:
We don't have any extra information. I think it's up to the user to decide what is and what isn't a realistic scenario (state), assuming that it is actually possible in Juju. I think you could have this in Juju but it would be a somewhat broken state to be in, so the person writing the tests probably has made a mistake, but we're not 100% sure on that so it's better to warn than error out.